lib/blockscore/connection.rb

```diff
@@ -0,0 +1,65 @@
+require 'json'
+require 'active_support/core_ext/module/attribute_accessors'
+require 'active_support/core_ext/string/inflections'
+require 'httparty'
+require 'blockscore/responder'
+require 'blockscore/errors/api_error'
+require 'blockscore/errors/authentication_error'
+require 'blockscore/errors/blockscore_error'
+require 'blockscore/errors/invalid_request_error'
+
+module BlockScore
+  module Connection
+    include BlockScore::Responder
+    mattr_accessor :api_key
+
+    def get(path, params)
+      request :get, path, params
+    end
+
+    def post(path, params)
+      request :post, path, params
+    end
+
+    def patch(path, params)
+      request :patch, path, params
+    end
+
+    def put(path, params)
+      request :put, path, params
+    end
+
+    def delete(path, params)
+      request :delete, path, params
+    end
+
+    private
+
+    def headers
+      @@headers ||= {
+        'Accept' => 'application/vnd.blockscore+json;version=4',
+        'User-Agent' => 'blockscore-ruby/4.1.0 (https://github.com/BlockScore/blockscore-ruby)',
+        'Content-Type' => 'application/json'
+      }
+    end
+
+    def request(method, path, params)
+      if @@api_key.nil?
+        fail BlockScore::AuthenticationError, { :message =>
```

Comment:
Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.<br>Style/HashSyntax: Use the new Ruby 1.9 hash syntax.